meson build fails: "too many decimal points in number ... in expansion of macro 'IMV_VERSION'"

I can't get this built with `nix` for `nixpkgs`.

This part looks weird to me:  `'-DIMV_VERSION=""4.1.0""' `.

```
these derivations will be built:
  /nix/store/x0bfl116ilbg42vb6vldihnncfvwfd5m-imv-04ef9440fb079736020d796142de18008da3ba62.drv
building '/nix/store/x0bfl116ilbg42vb6vldihnncfvwfd5m-imv-04ef9440fb079736020d796142de18008da3ba62.drv'...
unpacking sources
unpacking source archive /nix/store/vj39vpy1ciyry5svgfp6sfn8paqvv751-source
source root is source
patching sources
configuring
meson flags: --buildtype=plain         --libdir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/lib --libexecdir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/libexec         --bindir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/bin --sbindir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/sbin         --includedir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/include         --mandir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/share/man --infodir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/share/info         --localedir=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/429sd9z2rd34y4fzhm1340ssngrs71j3-imv-04ef9440fb079736020d796142de18008da3ba62 -Dlibnsgif=disabled
The Meson build system
Version: 0.52.1
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: imv
Project version: 4.1.0
C compiler for the host machine: /nix/store/w24ln4ja8i3xxfy63g9hqsnyfq4kq9mj-gcc-wrapper-9.2.0/bin/cc (gcc 9.2.0 "gcc (GCC) 9.2.0")
C linker for the host machine: GNU ld.bfd 2.31.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program git found: NO
Found pkg-config: /nix/store/zr0ick1qd775xmqp64c7cac8vsdff5sf-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Run-time dependency pangocairo found: YES 1.44.7
Run-time dependency gl found: YES 19.3.2
Run-time dependency threads found: YES
Run-time dependency xkbcommon found: YES 0.8.4
Run-time dependency icu-io found: YES 64.2
Run-time dependency wayland-client found: YES 1.17.0
Run-time dependency wayland-egl found: YES 18.1.0
Run-time dependency egl found: YES 19.3.2
Library rt found: YES
Run-time dependency x11 found: YES 1.6.8
Dependency gl found: YES (cached)
Run-time dependency glu found: YES 9.0.1
Run-time dependency xcb found: YES 1.13.1
Run-time dependency xkbcommon-x11 found: YES 0.8.4
Library freeimage found: YES
Library tiff found: YES
Run-time dependency libpng found: YES 1.6.37
Run-time dependency libturbojpeg found: YES 2.0.3
Run-time dependency librsvg-2.0 found: YES 2.46.4
Dependency libnsgif skipped: feature libnsgif disabled
Run-time dependency cmocka found: YES 1.1.5
Program a2x found: YES (/nix/store/kamj6pj941p77kkvr3mj0v72jj17lwg2-asciidoc-8.6.9/bin/a2x)
Message:

Building imv 4.1.0

Window systems enabled:
- wayland
- x11

Backends enabled:
- freeimage
- libtiff
- libpng
- libjpeg
- librsvg

Build targets in project: 8
Option buildtype is: plain [default: debugoptimized]
Found ninja-1.9.0 at /nix/store/i2xn0gms4wmv5mmvs5hw43mkmy5pzibx-ninja-1.9.0/bin/ninja
meson: enabled parallel building
building
build flags: -j8 -l8
[8/109] Compiling C object 'imv-msg@exe/src_imv.c.o'.K
FAILED: imv-msg@exe/src_imv.c.o
/nix/store/w24ln4ja8i3xxfy63g9hqsnyfq4kq9mj-gcc-wrapper-9.2.0/bin/cc -Iimv-msg@exe -I. -I.. -I/nix/store/shb0k6nar3wvczkp56pk9lz7f25sbjpx-glib-2.62.4-dev/include -I/nix/store/shb0k6nar3wvczkp56pk9lz7f25sbjpx-glib-2.62.4-dev/include/glib-2.0 -I/nix/store/0fsfl5zdz81d7042dpmch329q17cakvf-glib-2.62.4/lib/glib-2.0/include -I/nix/store/0lm1j37h1vrkfj1nr4y1gy67j9g53g70-cairo-1.16.0-dev/include/cairo -I/nix/store/rmi8bx6pysg7r82v7p77ahxz37rwwqh1-freetype-2.10.1-dev/include/freetype2 -I/nix/store/rmi8bx6pysg7r82v7p77ahxz37rwwqh1-freetype-2.10.1-dev/include -I/nix/store/i12x6b8qdrwrhxvw1a28im6lbzs52kf4-pango-1.44.7-dev/include/pango-1.0 -I/nix/store/xl31gc3009cdn5m6293l35662jvnmw1r-harfbuzz-2.6.4-dev/include/harfbuzz -I/nix/store/xjj17y0jl23zima2vjl0mx3cz72lj2w9-mesa-19.3.2-dev/include -I/nix/store/bdmrkq9r5qwzs8rv4aasn93fch3h97zh-libglvnd-1.2.0-dev/include -I/nix/store/pcflkzb5kiigbh5sb1ajvr7yaygczvw3-libxkbcommon-0.8.4-dev/include -I/nix/store/iwca3zfplqmimj54ajwr5gx6dfsnl3r0-icu4c-64.2-dev/include -I/nix/store/zwlzd0c6adm3v2az2dr13r3i4zjq2vwb-zlib-1.2.11-dev/include -I/nix/store/ngbg4r9amd2igbv4r5xpa21ygpx8fifi-libpng-apng-1.6.37-dev/include/libpng16 -I/nix/store/2wjhzmfh9b42nhafdwxx16q1fm5sikqi-libjpeg-turbo-2.0.3-dev/include -I/nix/store/3angsr986myign352vlhpaswgp00sppq-librsvg-2.46.4-dev/include/librsvg-2.0 -I/nix/store/hadkjwqirgs1n1ipnq12xaz38zx5lm1p-gdk-pixbuf-2.40.0-dev/include/gdk-pixbuf-2.0 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=c99 '-DIMV_VERSION=""4.1.0""' -D_XOPEN_SOURCE=700 -DIMV_BACKEND_FREEIMAGE -DIMV_BACKEND_LIBTIFF -DIMV_BACKEND_LIBPNG -DIMV_BACKEND_LIBJPEG -DIMV_BACKEND_LIBRSVG -pthread -MD -MQ 'imv-msg@exe/src_imv.c.o' -MF 'imv-msg@exe/src_imv.c.o.d' -o 'imv-msg@exe/src_imv.c.o' -c ../src/imv.c
../src/imv.c: In function 'print_help':
<command-line>: error: too many decimal points in number
../src/imv.c:713:57: note: in expansion of macro 'IMV_VERSION'
  713 |   printf("imv %s\nSee manual for usage information.\n", IMV_VERSION);
      |                                                         ^~~~~~~~~~~
<command-line>: error: expected ')' before numeric constant
../src/imv.c:713:57: note: in expansion of macro 'IMV_VERSION'
  713 |   printf("imv %s\nSee manual for usage information.\n", IMV_VERSION);
      |                                                         ^~~~~~~~~~~
../src/imv.c: In function 'imv_parse_args':
<command-line>: error: too many decimal points in number
../src/imv.c:757:33: note: in expansion of macro 'IMV_VERSION'
  757 |         printf("Version: %s\n", IMV_VERSION);
      |                                 ^~~~~~~~~~~
<command-line>: error: expected ')' before numeric constant
../src/imv.c:757:33: note: in expansion of macro 'IMV_VERSION'
  757 |         printf("Version: %s\n", IMV_VERSION);
      |                                 ^~~~~~~~~~~
../src/imv.c: In function 'command_exec':
../src/imv.c:1569:3: warning: ignoring return value of 'system', declared with attribute warn_unused_result [-Wunused-result]
 1569 |   system(argstr);
      |   ^~~~~~~~~~~~~~
[9/109] Compiling C object 'imv-msg@exe/src_image.c.o'.
../src/image.c: In function 'imv_image_free':
../src/image.c:52:5: warning: 'rsvg_handle_close' is deprecated: Use 'rsvg_handle_read_stream_sync' instead [-Wdeprecated-declarations]
   52 |     rsvg_handle_close(image->svg, &error);
      |     ^~~~~~~~~~~~~~~~~
In file included from ../src/image.h:7,
                 from ../src/image.c:1:
/nix/store/3angsr986myign352vlhpaswgp00sppq-librsvg-2.46.4-dev/include/librsvg-2.0/librsvg/rsvg.h:192:14: note: declared here
  192 | gboolean     rsvg_handle_close  (RsvgHandle *handle, GError **error);
      |              ^~~~~~~~~~~~~~~~~
[11/109] Compiling C object 'imv-msg@exe/src_keyboard.c.o'.
../src/keyboard.c: In function 'imv_keyboard_describe_key':
../src/keyboard.c:117:32: warning: '%s' directive output may be truncated writing up to 127 bytes into a region of size between 112 and 128 [-Wformat-truncation=]
  117 |   snprintf(buf, sizeof buf, "%s%s", prefix, keyname);
      |                                ^~           ~~~~~~~
In file included from /nix/store/8pp3i3hcp7bv0f8jllzqq7gcp9dbzvp9-glibc-2.27-dev/include/stdio.h:862,
                 from ../src/keyboard.c:5:
/nix/store/8pp3i3hcp7bv0f8jllzqq7gcp9dbzvp9-glibc-2.27-dev/include/bits/stdio2.h:64:10: note: '__builtin___snprintf_chk' output between 1 and 144 bytes into a destination of size 128
   64 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   65 |        __bos (__s), __fmt, __va_arg_pack ());
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[15/109] Compiling C object 'imv-msg@exe/src_navigator.c.o'.
ninja: build stopped: subcommand failed.
builder for '/nix/store/x0bfl116ilbg42vb6vldihnncfvwfd5m-imv-04ef9440fb079736020d796142de18008da3ba62.drv' failed with exit code 1
error: build of '/nix/store/x0bfl116ilbg42vb6vldihnncfvwfd5m-imv-04ef9440fb079736020d796142de18008da3ba62.drv' failed
```

--- a/meson.build
+++ b/meson.build
@@ -7,7 +7,7 @@ project(
   default_options: ['buildtype=debugoptimized', 'c_std=c99'],
 )
 
-version = '"@0@"'.format(meson.project_version())
+version = '@0@'.format(meson.project_version())
 prog_git = find_program('git', required: false)
 if prog_git.found()
   git_description = run_command([prog_git.path(), 'describe', '--dirty', '--always', '--tags'])
@@ -102,7 +102,7 @@ files_msg = files('src/imv_msg.c', 'src/ipc_common.c')
 enabled_backends = []
 foreach backend : [
   ['freeimage', 'library', 'freeimage'],
-  ['libtiff', 'library', 'tiff'],
+  ['libtiff', 'dependency', 'libtiff-4', []],
   ['libpng', 'dependency', 'libpng', []],
   ['libjpeg', 'dependency', 'libturbojpeg', []],
   ['librsvg', 'dependency', 'librsvg-2.0', '>= 2.44'],

